src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -171,48 +217,78 @@ We support
                         });
                     }
                 }
+            }
 
-                IEnumerable<IConfigurationSection> variantSections = configurationSection.GetSection(FeatureVariantsSectionName).GetChildren();
+            return new FeatureFlagDefinition()
+            {
+                Name = configurationSection.Key,
+                EnabledFor = enabledFor,
+            };
+        }
+
+        private DynamicFeatureDefinition ReadDynamicFeatureDefinition(IConfigurationSection configurationSection)
+        {
+            Debug.Assert(configurationSection != null);
```

Comment:
>Debug.Assert(configurationSection != null); [](http://example.com/codeflow?start=12&length=43)

Same here. Null check may not even be necessary since the calling methods already check for null before invoking these two methods.

Reply:
We often `Debug.Assert` use `Debug.Assert` in our private methods. It offers a couple benefits.

1. It makes it clear that the assertion can be relied on in the rest of the method execution. In this case, the reader will know the `configurationSection` is not null. 
2. When we look at this method we know that if we're calling it from another method we shouldn't pass null, and vice versa we can look at the places that call this and confirm that they're doing a null check, like you mentioned, before calling. 
3. If it is ever null for some reason, it will throw in debug mode so it can be fixed.